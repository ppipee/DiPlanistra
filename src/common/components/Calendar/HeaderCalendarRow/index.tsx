import React, { memo } from 'react'

import { Cell, CellWrapper } from '../components/CalendarCell/styled'
import { RowContainer } from '../components/CalendarRow/styled'
import { DAYS } from '../constants'

const HeaderCalendarRow = () => {
	return (
		<RowContainer>
			{DAYS.map((day) => (
				<CellWrapper key={`header-${day}`}>
					<Cell className="heading4">{day}</Cell>
				</CellWrapper>
			))}
		</RowContainer>
	)
}

export default memo(HeaderCalendarRow)
