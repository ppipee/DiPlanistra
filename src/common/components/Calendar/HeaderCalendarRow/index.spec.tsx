import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import HeaderCalendarRow from '.'

describe('HeaderCalendarRow', () => {
	it('should render HeaderCalendarRow correctly', () => {
		const { container } = render(<HeaderCalendarRow />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
