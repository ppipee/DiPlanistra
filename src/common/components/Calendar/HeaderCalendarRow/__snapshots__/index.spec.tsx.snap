// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCalendarRow should render HeaderCalendarRow correctly 1`] = `
.c2 {
  width: inherit;
  height: inherit;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  position: absolute;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  cursor: pointer;
  color: #7D8391;
}

.c2:hover {
  background-color: #CEF4CE;
}

.c1 {
  width: 32px;
  height: 32px;
  position: relative;
  padding: 0 4px;
  box-sizing: content-box;
}

.c1 > div {
  pointer-events: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      Su
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      Mo
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      Tu
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      We
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      Th
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      Fr
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2 heading4"
    >
      Sa
    </div>
  </div>
</div>
`;
